src/comb_filter.rs

```diff
@@ -19,28 +28,89 @@ pub enum Error {
     InvalidValue { param: FilterParam, value: f32 }
 }
 
+
+
 impl CombFilter {
     pub fn new(filter_type: FilterType, max_delay_secs: f32, sample_rate_hz: f32, num_channels: usize) -> Self {
-        todo!("implement")
+        let filter_params = (1.0, 0.0);
+        let delay_time = (max_delay_secs * sample_rate_hz) as usize;
+        let delay_line = vec![0.0; delay_time];
+
+        return CombFilter { filter_type, filter_params,max_delay_secs,sample_rate_hz, num_channels,delay_line };
     }
+    
 
     pub fn reset(&mut self) {
-        todo!("implement")
+        self.filter_type = FilterType::FIR; // Replace with the appropriate default value
```

Comment:
My thoughts for the reset function were that we want to restore the filter to its original settings before being altered by the set_param function. If we can only change delay and gain using set_param, then changing the filter type, max delay, sample rate, and number of channels to be anything but the values set in the new function seems unnecessary.